include/envoy/server/overload_manager.h

```diff
@@ -29,6 +30,23 @@ class OverloadActionState {
 
   float value() const { return action_value_; }
   bool isSaturated() const { return action_value_ == 1; }
+  bool isInactive() const { return action_value_ == 0; }
+
+  /**
+   * Converts the overload action state to a boolean. The return value will be true with probability
+   * of the level of saturation.
+   */
+  bool isRandomizedActive(Random::RandomGenerator& random_generator) const {
```

Comment:
nit: I'm unsure about the name of this method but not sure what would be a better name, possibly isPartiallySaturated

Reply:
From a public interface perspective, why isn't all of this just hidden by `isSaturated()`? Should the caller care about scaling if the result ends up being a bool? Also, instead of passing the random generator to this method can it just be passed to the constructor of the state or gotten from the `Api` after the recent @akonradi refactor?

Reply:
I can see some callers wanting to react only on the fully saturated state, but it would make sense to force those cases to access value() directly.  The definition of isSaturated could be as you describe and use an internal random generation to turn value() into a bool

Reply:
OverloadActionState is a value type right now (basically just a wrapper around the float); I'd be hesitant to start threading references into it.

Reply:
I will admit that I don't know the code structure and would have to research it, but it seems like conceptually we should be able to query the state via one function. If that returns an enum (not saturated, partially saturated, saturated) that's fine, but it seems like all of the randnomness, etc. should be abstracted away. How this is called from the HCM feels not right to me. Can we refactor in some way to better abstract it?

Reply:
Yeah, I think this is a utility function masquerading as a method. The state is available via `value()` and `isSaturated()` provides a convenient boolean accessor. What about making this a separate free function, or a method on RandomGenerator, that takes a float (result of `.value()`) and has the same logic?

Reply:
I would need to spend more time with the code to have a more formed opinion, so will defer to all of you on how to better name/expose this.

Reply:
I removed this method and added a RandomGenerator::bernoulli helper method inspired by [absl::Bernoulli](https://github.com/abseil/abseil-cpp/blob/c678d6c6bf70d47b4aa5bc3576a3a769775bc162/absl/random/distributions.h#L221) to use instead.